source/dofs/dof_renumbering.cc

```diff
@@ -406,25 +411,50 @@ namespace DoFRenumbering
     // note that if constraints are not requested, then the 'constraints'
     // object will be empty and using it has no effect
     IndexSet locally_relevant_dofs;
-    DoFTools::extract_locally_relevant_dofs(dof_handler, locally_relevant_dofs);
+    IndexSet const *ptr_locally_owned_dofs;
+    if (reorder_level_dofs == false)
+      {
+        DoFTools::extract_locally_relevant_dofs(dof_handler,
+                                                locally_relevant_dofs);
+        ptr_locally_owned_dofs = &dof_handler.locally_owned_dofs();
+      }
+    else
+      {
+        Assert(dof_handler.n_dofs(level) != numbers::invalid_dof_index,
+               ExcDoFHandlerNotInitialized());
+        DoFTools::extract_locally_relevant_level_dofs(dof_handler,
+                                                      level,
+                                                      locally_relevant_dofs);
+        ptr_locally_owned_dofs = &dof_handler.locally_owned_mg_dofs(level);
+      }
```

Comment:
I would just initialize via an immediately invoked lambda like
```suggestion
    const IndexSet = [&]() {
      if (reorder_level_dofs == false)
        {
          DoFTools::extract_locally_relevant_dofs(dof_handler,
                                                  locally_relevant_dofs);
          return dof_handler.locally_owned_dofs();
        }
      else
        {
          Assert(dof_handler.n_dofs(level) != numbers::invalid_dof_index,
                 ExcDoFHandlerNotInitialized());
          DoFTools::extract_locally_relevant_level_dofs(dof_handler,
                                                        level,
                                                        locally_relevant_dofs);
          return dof_handler.locally_owned_mg_dofs(level);
        }
      }();
```